homeassistant/components/homewizard/repairs.py

```diff
@@ -0,0 +1,74 @@
+"""Repairs platform for the HomeWizard integration."""
+from __future__ import annotations
+
+from typing import cast
+
+from homewizard_energy import HomeWizardEnergy
+from homewizard_energy.errors import DisabledError
+import voluptuous as vol
+
+from homeassistant import data_entry_flow
+from homeassistant.components.repairs import RepairsFlow
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_DEVICE_NAME
+
+
+class HomeWizardEnergyEnableApi(RepairsFlow):
+    """Handler for an issue fixing flow."""
+
+    def __init__(self, hass: HomeAssistant, host: str | None, name: str | None) -> None:
+        """Set variabled for repair context."""
+        self.host = host
+        self.name = name
+
+    async def async_step_init(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the first step of a fix flow."""
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the confirm step of a fix flow."""
+        if user_input is not None:
+            try:
+                api = HomeWizardEnergy(
+                    self.host, clientsession=async_get_clientsession(self.hass)
+                )
+                await api.device()
+            except DisabledError:
+                return self.async_abort(
+                    reason="api_not_enabled"
+                )  # Gives a blank "All OK"
```

Comment:
I still can't see how to tell the user that the repair has been failed. If the repair fails I just get a popup and the repair is removed:
<img width="595" alt="Screenshot 2022-08-05 at 14 20 52" src="https://user-images.githubusercontent.com/74970928/183083119-2f56f363-69e2-4c3c-bb59-3a8b2794ccce.png">